hal/inc/ble_hal.h

```diff
@@ -100,6 +100,12 @@ typedef enum hal_ble_scan_fp_t {
                                                                    address is a resolvable private address that cannot be resolved. */
 } hal_ble_scan_fp_t;
 
+typedef enum hal_ble_scan_phys_t {
+    BLE_SCAN_PHYS_1MBPS                         = 0x01,  /**< Standard 1MBPS format compatible with all BLE 4/5 devices */
+    BLE_SCAN_PHYS_CODED                         = 0x04,  /**< Coded Phy, 125 kbps, gives longer-range; BLE 5 only */
+    BLE_SCAN_PHYS_BOTH                          = 0x05,  /**< Simultaneous scan for 1MBPS and CODED phys formats */
```

Comment:
Since this file is shared for all platforms, including the future platform that might use a new BLE chip, these enum values don't have to be the same as what Nordic defines. Also they don't have to be bit masks and can start from `0x01` and increase by 1. Then implement an internal function like `toPlatformBlePhys()` to convert the HAL enum values to target platform defined values. Please refer to `toPlatformIoCaps()` in ble_hal.cpp.

Reply:
OK, enums are updated to be generic, not Nordic-specific. Nordic-specific values are only in toPlatformScanParams() function now.